### PR TITLE
draft release plan

### PR DESCRIPTION
**Pull request by bwhm**
WIP release plan.

### CODE REVIEW

**Comment by mnaamani** [comment]
## Runtime

- **Current version:** v5.3.0
- **New version:** target v5.4.0 (unless there are bugfix runtime upgrades before release)
- **Audit:** No
- **Documentation:** Publish the rust docs for the runtime at testnet.joystream.org/runtime-docs/ ?
- **Legal Review/ToS update:** No
- **Build/CI system:**
  - **CI**: Simple travis job for PRs, running cargo tests, and verifying build doesn't fail and rustfmt is used to format code
  - **Build**: Will have a working Docker file for building proposed WASM runtime blob 
- **Target Platforms:** WASM
-  **Substrate API:** substrate v1.0 branch - at commit `6dfc3e8b` (unchanged from v5.3.0)
- **New/Altered Functionality:**
    * New Forum module
    * Updated Actors module to support storage tranches
    * Updated storage modules to support storage tranches
    * Cleanup old migration code in members module
- **Refactor/Reorganization:**
  - Best effort should be made to make new runtime modules as separate git repos. [template](https://github.com/Joystream/runtime-example-module)
  - Existing modules can remain in same repo

## Runtime Proposal

- **Description:** 
     Create a council runtime upgrade `proposal` for a new runtime with a member key.
     provide a script/instructions for how to build the identical runtime proposed.
- **Time line:** After the runtime upgrade proposal is submitted, the actual upgrade will happen after all council members have voted.

## Public Infrastructure

### Faucet service backend
No changes from me other than perhaps changing name of testnet on the frontend from Athens to Acropolis.

### Hosted Joystream Pioneer

- **Description:** Host a version of Joystream Pioneer on testnet.joystream.org
- **Manager:** Mokhtar
- **DevOps:** Mokhtar
- **Repo:** (apps)[https://github.com/Joystream/apps] joystream branch (static build with yarn build)
- **Team:**
  - Mokhtar
- **Tasks:**
  - Update Caddy file to redirect https://testnet.joystream.org/athens/pioneer/ to https://testnet.joystream.org/acropolis/pioneer/


I don't think we will have time to build `Storage & distribution error endpoint` under public infrastructure 


### Hosted Joystream Storage Node - 

We may actually need to host more than one depending on how many tranches we choose to make available at launch.

- **Description:** A storage node without any special status
- **Manager:** **Jens**
- **DevOps:** **Jens** **Mokhtar**
- **Repo:** [storage-node-joystream](https://github.com/Joystream/storage-node-joystream)
- **Team:**
  - **Jens**
  - **Mokhtar**
- **Tasks:**
  - Host a node and signup to role in a tranche

File: okrs/README.md [hunk 1]
```diff
@@ -119,43 +119,40 @@ Archived OKRs are found in [archive](/OKR-archive), below only live OKRs are fou
...
-    - Bedeho: n_b/(n*2)    
-3. `Add tranches to storage-node (ewd)`
-    - Jens (2/3)
-    - Mokhtar (1/3)
-4. `No PRs merged to master (excluding bugfixes and "pioneer") after "Module Test" (ewd)`
+2. `Forum (runtime), storage (runtime and P2P) fully specd (n)`
+    - Bedeho (as manager): 1/2
+    - Bedeho (as writer): 1/6
+    - Jens: 2/6
+3. `Have 4x coverage for all 3 tranches on storage node (ewd)`
```
Comment (jfinkhaeuser) [inline okrs/README.md]: Before I got sick, we were discussing how feasible tranches are going to be if we drop the current hyperdrive backend and hyperdrive based synchronization for Acropolis, so I'm not sure that's going to stay. We'll discuss later, I suppose.

**Comment by bwhm** [comment]
@siman @mnaamani

File: testnets/acropolis/README.md [hunk 2]
```diff
@@ -141,124 +147,140 @@ The following public products will be part of this release.
...
+- **Legal Review/ToS update:** No
+- **Build/CI system:**
+  - **Jens**
+    * CI: Simple travis job, running unit tests.
+    * Build docker image
+- **Target Platforms:** Linux
+- **New/Altered Functionality:**
+  - Support storage tranches. The main difference is to stake for joining a tranche rather than the storage provider role.
+    - Allow multiple keys, or allow one key to stake for multiple tranches.
+  - Drop hyperdrive backend. Replace with file system or IPFS backend **TBD**
```
Comment (mnaamani) [inline testnets/acropolis/README.md]: We have decided on IPFS so we can remove TBD.

File: testnets/acropolis/README.md [hunk 3]
```diff
@@ -141,124 +147,140 @@ The following public products will be part of this release.
...
+    - Drop multiple repository concept, and repository IDs in URLs.
+    - Replace sync protocol stack with simpler synchronization, downloading or pinning content from staked tranches.
+  - Finalize DHT for liaison/download node selection
```
Comment (mnaamani) [inline testnets/acropolis/README.md]: We started with bittorrent-dht library we can switch to using the IPFS dht, and utilize IPNS in addition to basic key/values

Comment (mnaamani) [reply testnets/acropolis/README.md]: Additionally we will now naturally use IPFS based content ids, (the content id can be generated in browser using ipfs-js)

**Pull request merged**